python_ta/typecheck/base.py

```diff
@@ -217,6 +217,71 @@ def unify_call(self, func_type, *arg_types):
             self.unify(arg_type, param_type)
         return self._type_eval(new_func_type.__args__[-1])
 
+    def least_general_unifier(self, t1, t2):
+        if isinstance(t1, TypeVar) and isinstance(t2, TypeVar):
+            i1 = self._find(t1)
+            i2 = self._find(t2)
+            if issubclass(i1, i2):
+                return i2
+            elif issubclass(i2, i1):
+                return i1
+            else:
+                return Any
+        elif isinstance(t1, TypeVar):
+            i1 = self._find(t1)
+            if issubclass(i1, t2):
+                return t2
+            elif issubclass(t2, i1):
+                return i1
+            else:
+                return Any
+        elif isinstance(t2, TypeVar):
+            self.least_general_unifier(t2, t1)
```

Comment:
Many parts of your code here are weird because there aren't any return statements. The whole point of this method is to return something.